### PR TITLE
Python: Model outgoing HTTP client requests

### PR DESCRIPTION
**Pull request by RasmusWL**
I think this is a great starting point for modeling outgoing HTTP client requests. I wouldn't say it's perfect, but it should be good enough to get me going for the SSRF query I'm trying to get finished :smile:

### CODE REVIEW

File: python/ql/src/semmle/python/web/client/StdLib.qll [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    HttpConnectionHttpRequest() {
+        exists(ClassValue cls, AttrNode call_origin, Value constructor_call_value |
+            cls = httpConnectionClass() and
+            func = cls.lookup("request") and
+            this = func.getACall() and
+            this.getFunction().pointsTo(_, _, call_origin) and
+            call_origin.getObject().pointsTo(_, constructor_call_value, constructor_call) and
+            cls = constructor_call_value.getClass() and
+            constructor_call = cls.getACall()
+        )
```
Comment (tausbn) [inline python/ql/src/semmle/python/web/client/StdLib.qll]: :thinking: This looks quite complicated. It's not immediately clear to be why the last two lines are needed. Do you get spurious instances if you omit those lines?

File: python/ql/test/library-tests/web/client/stdlib/ClientHttpRequests.ql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import python
```
Comment (tausbn) [inline python/ql/test/library-tests/web/client/stdlib/ClientHttpRequests.ql]: I'm slightly confused by the test setup. If you have tests in both `2` and `3`, then why have tests in the shared test directory? Are these not just duplicating the testing that goes on in `2` and `3` separately?

Ah, no. I see the point now: the shared tests are testing the use of `six` as a compatibility layer. Still, it bugs me a bit that all three of these tests are essentially identical, except for the imports.

File: python/ql/src/semmle/python/web/client/StdLib.qll (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/web/client/StdLib.qll]: Yes. I should have left a comment in there explaining why I needed to do that.

File: python/ql/test/library-tests/web/client/stdlib/ClientHttpRequests.ql (hunk 2 above)
Comment (RasmusWL) [reply python/ql/test/library-tests/web/client/stdlib/ClientHttpRequests.ql]: I reduced it to 2 now, which is slightly better. I still want a test that explicitly makes sure that `six` works, so personally I don't see how we can reduce it to 1. If you have ideas, I'm all ears

Comment (tausbn) [reply python/ql/test/library-tests/web/client/stdlib/ClientHttpRequests.ql]: I don't think we can reduce it to a single test, as we want to test three different things, but we only get two invocations of the test runner to do so. I think the combined test + tests specific to `six` makes sense as a division.

**Pull request merged**